OA-PASS/pass-docker
# Use authz 0.3.1

## Overview
* Updates `fcrepo` image to use new model-3.1 compatible User service
* Update the `authz` listener image to use the latest authz code as well.
* Updates the `ldap` and `idp` and `sp` images to provide `uinque-id`s
* Updates fcrepo and authz images to print out environment variables upon startup

## To test
There is no working ember image, so these are some steps you can do without Ember

* Try the user service (as various users?).  Hit `https://pass.local/pass-user-service/whoami` and log in to shib as one of the various test users.   (note:  for the first time, the likes of `staff2`, a staff member with no grants, will be let in).  Look at the `README.md` for a full list of plausible users.

* Try looking at submissions, and making them read-only.  They're read-only when submitted.   For example, `nih-user` has this incomplete submission as determined by the NIHMS etl: `https://pass.local/fcrepo/rest/submissions/04/4c/0a/2e/044c0a2e-43fa-4611-889a-36fd7c104161`.  Feel free to change this to a terminal state (`submitted=true`, or `submissionStatus={submitted, needs-attention, complete}`.    Reload after changing this, and the edit buttons should go away to indicate that it is read-only.
  * you can do this on the "Update Properties" panel on the side, using RDF.  Type in the triple you want to add and delete.  For example:
<pre>
    DELETE { &lt;&gt; pass:submissionStatus "manuscript-required" }
    INSERT { &lt;&gt; pass:submissionStatus "needs-attention" }
    WHERE { }
</pre>

* Logged in to shib as one of the various test users, try creating a submissionEvent, and verifying that it is immutable.  The easiest way to do this is to point the browser to `https://pass.local/fcrepo/rest/submissionEvents`, then under "Create a New Child Resource"   Select "container" from the menu, and type in any name under "identifier".    Then add an rdf type in the text box before clicking `add`:

<pre>
    &lt;&gt; a  &lt;http://oapass.org/ns/pass#SubmissionEvent&gt; .
</pre>

It'll put you on that submission event's page.  Click reload, and make sure all the edit/update boxes are gone (indicating it is read-only)

Resolves OA-PASS/pass-authz#57
